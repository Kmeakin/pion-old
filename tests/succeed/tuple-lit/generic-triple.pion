//~ run = "${PION} elab ${FILE}"

let get1 = fun (A: Type) (B: Type) (C: Type) (p: {_0: A, _1: B, _2: C}) => p._0;
let get2 = fun (A: Type) (B: Type) (C: Type) (p: (A, B, C)) => p._1;
let get3 = fun (A: Type) (B: Type) (C: Type) (p: (A, B, C)) => p._2;
{}
