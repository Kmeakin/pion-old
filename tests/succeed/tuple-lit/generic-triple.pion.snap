exit status: 0
stdout:
let get1 : fun (A : Type) (B : Type) (C : Type) -> (
    A,
    B,
    C,
) -> A = fun (A : Type) (B : Type) (C : Type) (p : (A, B, C)) => p._0;
let get2 : fun (A : Type) (B : Type) (C : Type) -> (
    A,
    B,
    C,
) -> B = fun (A : Type) (B : Type) (C : Type) (p : (A, B, C)) => p._1;
let get3 : fun (A : Type) (B : Type) (C : Type) -> (
    A,
    B,
    C,
) -> C = fun (A : Type) (B : Type) (C : Type) (p : (A, B, C)) => p._2;
() : ()

stderr:
